io-engine/tests/resource_stats.rs

```diff
@@ -0,0 +1,408 @@
+        .with_name("pool1")
+        .with_new_uuid()
+        .with_malloc("mem1", POOL_SIZE);
+
+    let mut repl_2 = ReplicaBuilder::new(ms_1.clone())
+        .with_pool(&pool_1)
+        .with_name("r1")
+        .with_new_uuid()
+        .with_thin(false)
+        .with_size_mb(REPL_SIZE);
+
+    pool_1.create().await.unwrap();
+    repl_2.create().await.unwrap();
+    repl_2.share().await.unwrap();
+
+    let mut nex_0 = NexusBuilder::new(ms_nex.clone())
+        .with_name("nexus0")
+        .with_new_uuid()
+        .with_size_mb(REPL_SIZE)
+        .with_replica(&repl_1)
+        .with_replica(&repl_2)
+        .with_replica(&repl_0);
+
+    nex_0.create().await.unwrap();
+    nex_0.publish().await.unwrap();
+
+    ms_nex.lock().await.stats.reset_io_stats(()).await.unwrap();
+    ms_0.lock().await.stats.reset_io_stats(()).await.unwrap();
+    ms_1.lock().await.stats.reset_io_stats(()).await.unwrap();
+
+    let pool_0_stat = ms_0
+        .lock()
+        .await
+        .stats
+        .get_pool_io_stats(ListStatsOption {
+            name: None,
+        })
+        .await
+        .unwrap();
+
+    let pool_1_stat = ms_1
+        .lock()
+        .await
+        .stats
+        .get_pool_io_stats(ListStatsOption {
+            name: None,
+        })
+        .await
+        .unwrap();
+
+    let nexus_stat = ms_nex
+        .lock()
+        .await
+        .stats
+        .get_nexus_io_stats(ListStatsOption {
+            name: None,
+        })
+        .await
+        .unwrap();
+
+    let repl_1_stat = ms_0
+        .lock()
+        .await
+        .stats
+        .get_replica_io_stats(ListStatsOption {
+            name: None,
+        })
+        .await
+        .unwrap();
+
+    let repl_2_stat = ms_1
+        .lock()
+        .await
+        .stats
+        .get_replica_io_stats(ListStatsOption {
+            name: None,
+        })
+        .await
+        .unwrap();
+
+    let repl_0_stat = ms_nex
+        .lock()
+        .await
+        .stats
+        .get_replica_io_stats(ListStatsOption {
+            name: None,
+        })
+        .await
+        .unwrap();
+
+    let nexus_stat: &Vec<IoStats> = &nexus_stat.get_ref().stats;
+    let pool_0_stat: &Vec<IoStats> = &pool_0_stat.get_ref().stats;
+    let pool_1_stat: &Vec<IoStats> = &pool_1_stat.get_ref().stats;
+    let repl_1_stat: &Vec<ReplicaIoStats> = &repl_1_stat.get_ref().stats;
+    let repl_2_stat: &Vec<ReplicaIoStats> = &repl_2_stat.get_ref().stats;
+    let repl_0_stat: &Vec<ReplicaIoStats> = &repl_0_stat.get_ref().stats;
+    let nexus_stat = nexus_stat.get(0).unwrap();
+    let pool_0_stat = pool_0_stat.get(0).unwrap();
+    let pool_1_stat = pool_1_stat.get(0).unwrap();
+    let repl_1_stat = repl_1_stat.get(0).unwrap();
+    let repl_2_stat = repl_2_stat.get(0).unwrap();
+    let repl_0_stat = repl_0_stat.get(0).unwrap();
+
+    // Validate num_read/write_ops reset across resource.
+    assert_eq!(repl_1_stat.stats.clone().unwrap().num_read_ops, 0);
```

Comment:
I think all these can be as_ref ?
```suggestion
    assert_eq!(repl_1_stat.stats.as_ref().unwrap().num_read_ops, 0);
```